### PR TITLE
Centos 8 fixes

### PR DESCRIPTION
**Pull request by meffie**

### CODE REVIEW

**Comment by meffie** [comment]
Cheyenne, can you verify this doesnt break your setup?

We should probably investigate if changes are needed to support selinux restricted mode on centos 8.

**Comment by CheyenneWills** [comment]
> Cheyenne, can you verify this doesnt break your setup?
> 
> We should probably investigate if changes are needed to support selinux restricted mode on centos 8.

I'll try to take a look this weekend

File: roles/openafs_server/tasks/main.yaml [hunk 1]
```diff
@@ -8,6 +8,12 @@
     - "../vars/{{ ansible_os_family }}.yaml"
     - "../vars/defaults.yaml"
 
+- name: Remove loopback address for our hostname in /etc/hosts
+  lineinfile:
+    state: absent
+    path: /etc/hosts
+    regexp: '^127\.\d+\.\d+\.\d+\s.*{{ ansible_host }}'
```
Comment (CheyenneWills) [inline roles/openafs_server/tasks/main.yaml]: Not quite sure you want to do exactly this.  Removing the entire localhost entry might cause grief.
You probably want to just remove the hostname and leave the "localhost" part alone..
e.g.
127.0.0.1 localhost myhost.mydomain.net  ==> 127.0.0.1 localhost

File: roles/openafs_robotest/tasks/main.yaml [hunk 2]
```diff
@@ -45,7 +45,7 @@
   become: yes
   become_user: "{{ afs_robotest_user }}"
   git:
-    repo: https://github.com/openafs-contrib/openafs-robotest.git
+    repo: "{{ afs_robotest_git_repo }}"
```
Comment (CheyenneWills) [inline roles/openafs_robotest/tasks/main.yaml:48]: +1

File: playbooks/testcell.yaml [hunk 3]
```diff
@@ -66,5 +66,26 @@
...
+    - test-setup
+
+- name: Run tests
+  hosts: afs_robotest[0]
   tags:
     - test
+  tasks:
+    - name: Run tests
+      command:
+        cmd: "/usr/bin/run-openafs-robotest.sh {{ afs_robotest_suite | d('tests') }}"
```
Comment (CheyenneWills) [inline playbooks/testcell.yaml:79]: May not have been installed in /usr/bin

**Review by CheyenneWills — COMMENTED** [review]
A couple of comments.  Probably the biggest one is about modifying /etc/hosts

File: roles/openafs_server/vars/CentOS-8.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+# to set the selinux context before bosserver starts.
+afs_selinux_top_dirs:
+ - "/usr/afs"
+ - "/usr/vice"
+
+# Default selinux mode.
+_afs_selinux_mode: permissive
+
+# Build default options
+_afs_server_configure_opts: >
```
Comment (CheyenneWills) [inline roles/openafs_server/vars/CentOS-8.yaml:41]: I believe this should be _afs_devel_configure_opts and should probably be in /roles/openafs_devel/vars/CentOS-8.yaml



I believe that things get set as follows in roles/openafs_devel/tasks/build.yaml  using the following hierarchy (first match "wins")

For server:
 afs_devel_server_configure_opts       (reserved for user to set)
 afs_devel_configure_opts                   (reserved for user to set)
_afs_devel_configure_opts                  this is reserved for "internal" role use.  But we need to see if a "calling" role (e.g. openafs_server / openafs_client) can override what is set in openafs_devel/vars.

Not sure if your intent is for  "openafs_devel" to have its own default configuration as well as allowing the openafs_server / openafs_client to have their own default configuration.  If so, there may need to be some more hackery in setting this.

**Review by CheyenneWills — CHANGES_REQUESTED** [review]

File: roles/openafs_client/vars/CentOS-8.yaml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+afs_client_service: openafs-client
+
+# OS dependent variables
+afs_firewall: firewalld
+
+# Install dkms by default in CentOS 8.
+_afs_client_install_dkms: yes
+
+# Build default options.
+_afs_client_configure_opts: >
```
Comment (CheyenneWills) [inline roles/openafs_client/vars/CentOS-8.yaml:16]: See the note attached to roles/openafs_server/vars/CentosOS-8.yaml in "Default selinux mode to permissive on CentOS 8"

**Review by CheyenneWills — CHANGES_REQUESTED** [review]

File: roles/openafs_server/tasks/main.yaml (hunk 1 above)
Comment (meffie) [reply roles/openafs_server/tasks/main.yaml]: Ah, interesting. On debian, a separate line is added,
127.0.1.1  myhost
127.0.0.1 localhost

Removing the "localhost" line would be problematic. I'll change this so we dont remove the localhost entry.

File: roles/openafs_server/vars/CentOS-8.yaml (hunk 4 above)
Comment (meffie) [reply roles/openafs_server/vars/CentOS-8.yaml:41]: Thanks

_afs_devel_configure_ops go in openafs_devel. the convention is '_afs_devel" ==> openafs_devel,  _afs_server => openafs_server, afs_client ==> openafs_client, etc.

File: roles/openafs_client/vars/CentOS-8.yaml (hunk 5 above)
Comment (meffie) [reply roles/openafs_client/vars/CentOS-8.yaml:16]: should be fixed with the openafs_devel change to add CentOS-8 there?

**Comment by meffie** [comment]
Thanks for the review.

**Comment by meffie** [comment]
I dropped the changes for debian in this pr. Will put those in a different pr. Thanks.

**Pull request merged**